### PR TITLE
Server side fitb

### PR DESCRIPTION
**Pull request by bjones1**
This PR allows grading of fill-in-the-blank questions on the server. However, server-side grading only occurs when a course is built with server-side grading enabled. Therefore, current behavior (client-side grading) will be maintained until a RunestoneComponents PR provides the necessary updates.

Merge #1168 before this.

### CODE REVIEW

**Comment by bjones1** [comment]
Fixed now. I've written lots of tests for this, but they won't pass until a RunestoneComponents PR is merged.

File: modules/rs_grading.py [hunk 1]
```diff
@@ -2,15 +2,12 @@
...
 
-# When testing, the ``settings`` object isn't defined. Import it in this case.
-try:
-    logger
-except:
-    execfile('applications/runestone/models/0.py', globals())
-logger = logging.getLogger(settings.logger)
-logger.setLevel(settings.log_level)
+logger = logging.getLogger(current.settings.logger)
+logger.setLevel(current.settings.log_level)
```
Comment (bnmnetp) [inline modules/rs_grading.py]: We should make a todo  that functions like _scorable_mchoice_answers no longer need to pass db as a parameter.

Comment (bjones1) [reply modules/rs_grading.py]: Good point. Do you want me to submit an issue? Or would you like to?

**Comment by bjones1** [comment]
That's correct -- unless server-side grading is explicitly enabled for a book by setting `runestone_server_side_grading` to `True` and all required conditions are met, this will do all grading client-side. The conditions are:

- The use of Runestone services (``eBookConfig.useRunestoneServices === true``)
- A valid login (``eBookConfig.loginRequired === true``)
- Logging enabled (``eBookConfig.logLevel  > 0``)

**Comment by bnmnetp** [comment]
I hear you.  I'll finish up my testing on this over the weekend.

**Comment by bjones1** [comment]
Great, thanks!

**Comment by bjones1** [comment]
Thanks!

**Pull request merged**